crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,217 @@
+use clap::Parser;
+use notify::{self, RecursiveMode, Watcher};
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use tokio::sync::mpsc;
+use tokio::sync::Mutex;
+use tokio::time;
+
+use crate::commands::build;
+
+use super::build::clients::LoamEnv;
+
+enum Message {
+    FileChanged,
+}
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub build_cmd: build::Cmd,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+fn canonicalize_path(path: &Path) -> PathBuf {
+    if path.as_os_str().is_empty() {
+        env::current_dir().unwrap_or_else(|_| PathBuf::from("."))
+    } else if path.components().count() == 1 {
+        // Path is a single component, assuming it's a filename
+        env::current_dir()
+            .unwrap_or_else(|_| PathBuf::from("."))
+            .join(path)
+    } else {
+        fs::canonicalize(path).unwrap_or_else(|_| path.to_path_buf())
+    }
+}
+
+fn is_parent_in_watched_dirs(parent: &Path, watched_dirs: &[Arc<PathBuf>]) -> bool {
+    watched_dirs.iter().any(|p| canonicalize_path(p) == parent)
+}
+
+fn is_temporary_file(path: &Path) -> bool {
+    let file_name = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+
+    // Vim temporary files
+    if file_name.starts_with('.') {
```

Comment:
This could also be an `.env` file. Do they end with `.swp`?

Reply:
`.swp` wasn't the only one but I think `.swo`, `~` and `.swp` should cover it all.